webapp/channels/src/components/new_search/new_search.tsx

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState, useRef, useCallback} from 'react';
+import {FormattedMessage} from 'react-intl';
+import {useSelector, useDispatch} from 'react-redux';
+import styled from 'styled-components';
+
+import {getCurrentChannelNameForSearchShortcut} from 'mattermost-redux/selectors/entities/channels';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+
+import {
+    updateSearchTerms,
+    showSearchResults,
+    updateSearchType,
+} from 'actions/views/rhs';
+import {
+    getSearchTerms,
+} from 'selectors/rhs';
+
+import Popover from 'components/widgets/popover';
+
+import Constants from 'utils/constants';
+import * as Keyboard from 'utils/keyboard';
+import {isServerVersionGreaterThanOrEqualTo} from 'utils/server_version';
+import {isDesktopApp, getDesktopVersion, isMacApp} from 'utils/user_agent';
+
+import type {GlobalState} from 'types/store';
+
+import SearchBox from './search_box';
+
+const PopoverStyled = styled(Popover)`
+    min-width: 600px;
+    left: -90px;
+    top: -12px;
+    border-radius: 12px;
+
+    .popover-content {
+        padding: 0px;
+    }
+`;
+
+const NewSearchContainer = styled.div`
+    display: flex;
+    position: relative;
+    align-items: center;
+    height: 28px;
+    width: 100%;
+    background-color: rgba(var(--sidebar-text-rgb), 0.08);
+    color: rgba(var(--sidebar-text-rgb), 0.64);
+    font-size: 12px;
+    font-weight: 500;
+    border-radius: var(--radius-s);
+    border: none;
+    padding: 4px;
+    cursor: pointer;
+    &:hover {
+        background-color: rgba(var(--sidebar-text-rgb), 0.16);
+        color: rgba(var(--sidebar-text-rgb), 0.88);
+    }
+`;
+
+const NewSearch = (): JSX.Element => {
+    const currentChannelName = useSelector(getCurrentChannelNameForSearchShortcut);
+    const searchTerms = useSelector(getSearchTerms) || '';
+    const license = useSelector(getLicense);
+    let pluginSearch = useSelector((state: GlobalState) => state.plugins.components.SearchButtons);
+    if (license.IsLicensed !== 'true') {
+        pluginSearch = [];
+    }
```

Comment:
This should be combined into a memoized selector or else `pluginSearch` will cause `runSearch` to be recreated every time this is run when the server is unlicensed. Something like
```
createSelector(
    getLicense,
    (state) => state.plugins.components.SearchButtons,
    ...
);
```
should do the trick

Reply:
I fixed it with useMemo, to not create another selector for this specific use case.